oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/FulltextIndexConstants.java

```diff
@@ -436,4 +437,12 @@ public static IndexingMode from(String indexingMode) {
      * Boolean property indicating if in-built analyzer should preserve original term
      */
     String INDEX_ORIGINAL_TERM = "indexOriginalTerm";
+
+    /**
+     * Internal version of the index definition for specific index type. Index version is an information that might be
+     * needed from an outside process that does not have visibility to the specific index module.
+     */
+    Map<String, String> INDEX_VERSION_BY_TYPE = Map.of(
+            "elasticsearch", "1.1.0"
```

Comment:
Here we are using an hardcoded string, but there is already a constant defined for this String in ElasticIndexDefinition. Since we are anyway exposing in oak-search the index type of Elastic, I think it would be better to move that constant somewhere to the oak-search module and use it here as well.

Reply:
I prefer to not do that (at least in this PR) since moving the Elastic index type in oak-search would require a big refactoring (I see 32 references). Moreover, we should deprecate it first and remove it in a future version.